google/cloud/pubsub/internal/extend_leases_with_retry.cc

```diff
@@ -0,0 +1,171 @@
+// Copyright 2022 Google LLC
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     https://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+#include "google/cloud/pubsub/internal/extend_leases_with_retry.h"
+#include "google/cloud/pubsub/internal/exactly_once_policies.h"
+#include "google/cloud/completion_queue.h"
+#include "google/cloud/log.h"
+#include <algorithm>
+#include <memory>
+
+namespace google {
+namespace cloud {
+namespace pubsub_internal {
+GOOGLE_CLOUD_CPP_INLINE_NAMESPACE_BEGIN
+namespace {
+
+// This value is hard-coded in the spec.
+auto constexpr kMaxRetryAttempts = 3;
```

Comment:
Is the spec: "try three times" or "retry three times" ?

If the former, I would think this value should be 2.

Reply:
Fixed.